src/solver/conv_wino_fury_RxS.cpp

```diff
@@ -0,0 +1,569 @@
+    f_strides.w          = is_forward ? f_strides.w : -f_strides.w;
+    uint32_t f_RS_offset = is_forward ? 0 : ((conv.R - 1) * conv.S + (conv.S - 1));
+
+    result.invoker_factory = [=](std::vector<Kernel> kernels) {
+        return [=](const Handle& handle, const AnyInvokeParams& primitive_params) {
+            const auto k = handle.Run(kernels[0]);
+            const auto data_tensors =
+                !is_backWrW ? primitive_params.CastTo<conv::DataInvokeParams>().tensors.in
+                            : primitive_params.CastTo<conv::WrWInvokeParams>().tensors.x;
+            const auto filter_tensors =
+                !is_backWrW ? primitive_params.CastTo<conv::DataInvokeParams>().tensors.w
+                            : primitive_params.CastTo<conv::WrWInvokeParams>().tensors.dy;
+            const auto out_tensors =
+                !is_backWrW ? primitive_params.CastTo<conv::DataInvokeParams>().tensors.out
+                            : primitive_params.CastTo<conv::WrWInvokeParams>().tensors.dw;
+
+            // clang-format off
+            MIOPEN_LOG_I2(
+              "\nn_groups=" << conv.n_groups <<
+              "\nout_Hs=" << conv.out_Hs <<
+              "\nout_We=" << conv.out_We <<
+              "\nd_W_window=" << conv.d_W_window <<
+              "\nd_H_window=" << conv.d_H_window <<
+              "\nd_W_clip_bot_neg=" << conv.d_W_clip_bot_neg <<
+              "\nd_H_clip_bot_neg=" << conv.d_H_clip_bot_neg <<
+              "\nd_strides.nk=" << d_strides.nk <<
+              "\nd_strides.h=" << d_strides.h <<
+              "\nd_strides.c=" << d_strides.c <<
+              "\no_W_window=" << conv.o_W_window <<
+              "\no_H_window=" << conv.o_H_window <<
+              "\no_W_clip_bot_neg=" << conv.o_W_clip_bot_neg <<
+              "\no_H_clip_bot_neg=" << conv.o_H_clip_bot_neg <<
+              "\no_strides.nk=" << o_strides.nk <<
+              "\no_strides.h=" << o_strides.h <<
+              "\no_strides.c=" << o_strides.c <<
+              "\nN=" << conv.N <<
+              "\nK=" << conv.K <<
+              "\nC=" << conv.C <<
+              "\nR=" << conv.R <<
+              "\nS=" << conv.S <<
+              "\nf_strides.nk=" << f_strides.nk <<
+              "\nf_strides.c=" << f_strides.c <<
+              "\nf_strides.h=" << f_strides.h <<
+              "\nf_strides.w=" << f_strides.w <<
+              "\nf_RS_offset=" << f_RS_offset <<
+              "\nflags=" << flags <<
+              "\nactivation_mode=" << static_cast<uint32_t>(activation_mode) <<
+              "\n");
+            // clang-format on
+
+            k(conv.n_groups,
+              conv.out_Hs,
+              conv.out_We,
+              conv.d_W_window,
+              conv.d_H_window,
+              conv.d_W_clip_bot_neg,
+              conv.d_H_clip_bot_neg,
+              uint32_t(d_strides.nk),
+              uint32_t(d_strides.h), // todo: check < 2^16
+              uint32_t(d_strides.c),
```

Comment:
Will the type conversion inside invocation lambda create an overhead on kernel launch? If yes, then move the type cast outside the lambda.

Reply:
It heavily depends on compiler - usually it just moves data from one memory location(usually on stack) to another location(on stack as well), and for this particular cast it should use just different flavors of `mov` instruction, like `mov64 rax,[mem]; mov32 [mem], eax` or `mov32 eax,[mem]; mov32 [mem], eax`.

But I'll double check if there are any potential perf problems.

Reply:
That's exactly the difference - it uses just different flavors of mov.